components/nimbus/src/lib.rs

```diff
@@ -2,620 +2,50 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+#[cfg(feature = "stateful")]
 mod behavior;
-mod dbcache;
-mod enrollment;
-pub mod error;
-mod evaluator;
-use behavior::EventStore;
-use chrono::{DateTime, NaiveDateTime, Utc};
-use defaults::Defaults;
-pub use error::{NimbusError, Result};
+#[cfg(feature = "stateful")]
 mod client;
 mod config;
+#[cfg(feature = "stateful")]
+mod dbcache;
 mod defaults;
+mod enrollment;
+mod evaluator;
 mod matcher;
-pub mod persistence;
 mod sampling;
 mod strings;
+#[cfg(feature = "stateful")]
 mod updating;
+
+pub mod error;
+#[cfg(feature = "stateful")]
+pub mod nimbus;
```

Comment:
I wonder if this the only thing that should be in this file.

The rest of this current file is schema declarations for interacting with remote settings, so perhaps should be in its own `schema.rs`. WDYT?

Reply:
That's a good idea!